src/x/pool/object.go

```diff
@@ -138,6 +148,15 @@ func (p *objectPool) Put(obj interface{}) {
 		p.onPoolAccessErrorFn(errPoolAccessBeforeInitialized)
 		return
 	}
+
+	if obj, ok := obj.(FinalizeableOnce); ok {
+		if obj.Finalized() {
+			p.onPoolAccessErrorFn(fmt.Errorf("double finalize, %v", obj))
```

Comment:
nit: `double finalize, %v` -> `double finalize: %v` as per other errors using colon between message and value.

Reply:
I would use `reflect.TypeOf(obj).String()` here. "%v" on `obj` will be fairly non-descript/not very useful.

Reply:
👍